numpy/testing/_private/utils.py

```diff
@@ -404,6 +389,31 @@ def assert_equal(actual, desired, err_msg='', verbose=True):
     except (TypeError, ValueError, NotImplementedError):
         pass
 
+    # Inf/nan/negative zero handling
+    try:
+        isdesnan = gisnan(desired)
+        isactnan = gisnan(actual)
+        if isdesnan and isactnan:
+            return  # both nan, so equal
+
+        # handle signed zero specially for floats
+        array_actual = array(actual)
+        array_desired = array(desired)
```

Comment:
nit: maybe `asarray`?

Reply:
There is a check above for `isinstance(actual, ndarray) or isinstance(desired, ndarray): return assert_array_equal(actual, desired, ...`. I copied the pattern from (after the changes) line 380 in the clause above this one, should I change them both?

Reply:
Fair enough, I suppose that means there's no point using asarray.